telematic_system/telematic_apps/web_app/server/file_upload_server.js

```diff
@@ -62,52 +85,86 @@ const postListener = async (req, res) => {
       formidable().parse(req, async (err, fields, files) => {
         await listAllFiles(req, res)
           .then((data) => {
-            res.writeHead(200);
-            res.write(JSON.stringify(data));
-            res.end();
+            sendResponse(res, data);
           })
           .catch((err) => {
-            res.writeHead(500);
-            res.write(
-              JSON.stringify({ error: err.message || "Unknown server error!" })
-            );
-            res.end();
+            serverError(res, err);
           });
       });
       break;
     case HTTP_URLS.API_FILE_UPLOAD:
       await uploadFile(req)
         .then((data) => {
-          res.writeHead(200);
-          res.write(JSON.stringify(data));
-          res.end();
+          sendResponse(res, data);
         })
         .catch((err) => {
-          res.writeHead(500);
-          res.write(
-            JSON.stringify({ error: err.message || "Unknown server error!" })
-          );
-          res.end();
+          serverError(res, err);
         });
       break;
+    case HTTP_URLS.API_FILE_DESCRIPTION_UPDATE:
+      formidable().parse(req, async (err, fields, files) => {
+        let fileInfo = JSON.parse(fields["fields"]);
+        await updateDescription(fileInfo)
+          .then((data) => {
+            sendResponse(res, data);
+          })
+          .catch((err) => {
+            serverError(res, err);
+          });
+      });
+      break;
+    case HTTP_URLS.API_FILE_PROCESS_REQUEST:
+      formidable().parse(req, async (err, fields, files) => {
+        try {
+          let fileInfo = JSON.parse(fields["fields"]);
+          //Send file process request to NATS
+          let processingReq = {
+            uploaded_path: uploadDestPath,
+            filename: fileInfo.original_filename,
+          };
+
+          let natsConn = await createNatsConn();
+          await pubFileProcessingReq(natsConn, processingReq);
```

Comment:
Now that we have added the nats request subscription on a separate thread for the rosbag_processing service, can we check for a response and print it to screen?

Reply:
yes, the response is printed.

Reply:
![image](https://github.com/usdot-fhwa-stol/cda-telematics/assets/62157949/05f92ed2-eddd-40aa-bec4-3a82baeca9a9)

Reply:
![image](https://github.com/usdot-fhwa-stol/cda-telematics/assets/62157949/bbe937c7-f79c-4ee4-80e7-9ca3ab6d10cd)